orderer/consensus/etcdraft/blockpuller.go

```diff
@@ -160,51 +157,49 @@ func NewBlockFetcher(support consensus.ConsenterSupport,
 		CensorshipSuspicionThreshold: time.Duration((int64(clusterConfig.ReplicationPullTimeout) * shuffleTimeoutPercentage / 100)),
 		PeriodicalShuffleInterval:    shuffleTimeoutMultiplier * clusterConfig.ReplicationPullTimeout,
 		MaxRetries:                   uint64(clusterConfig.ReplicationMaxRetries),
-		MaxByzantineNodes:            maxByzantineNodes,
 	}
 
-	// TODO: Should this func be moved to `orderer/common/cluster` package
-	blockVerifierFunc := func(header *common.BlockHeader, metadata *common.BlockMetadata) error {
-		return verifyBlockSequence([]*common.Block{
-			{
-				Header:   header,
-				Metadata: metadata,
-			},
-		}, support.ChannelID())
+	bf_logger := flogging.MustGetLogger("orderer.common.cluster.puller").With("channel", support.ChannelID())
+
+	lastConfigBlock, err := lastConfigBlockFromSupport(support)
+	if err != nil {
+		return nil, err
 	}
 
+	verifierFactory := cluster.BlockVerifierBuilder(bccsp)
+
 	bf := cluster.BlockFetcher{
-		FetcherConfig:   fc,
-		LastConfigBlock: &common.Block{},
-		BlockVerifierFactory: func(block *common.Block) cluster.BlockVerifierFunc {
-			// block is a config block
-			return blockVerifierFunc
-		},
-		VerifyBlock: blockVerifierFunc,
+		FetcherConfig:        fc,
+		LastConfigBlock:      lastConfigBlock,
+		BlockVerifierFactory: verifierFactory,
+		VerifyBlock:          verifierFactory(lastConfigBlock),
 		AttestationSourceFactory: func(c cluster.FetcherConfig, latestConfigBlock *common.Block) cluster.AttestationSource {
-			// TODO: update FetcherConfig from latestConfigBlock
+			fc, err := cluster.UpdateFetcherConfigFromConfigBlock(c, latestConfigBlock)
+			bf_logger.Errorf("Could not update FetcherConfig fom Config Block: %v", err)
 			return &cluster.AttestationPuller{
 				Config: fc,
-				Logger: flogging.MustGetLogger("orderer.common.cluster.attestationpuller").With("channel", support.ChannelID()),
+				Logger: flogging.MustGetLogger("orderer.common.cluster.attestationpuller").With("channel", fc.Channel),
 			}
 		},
 		BlockSourceFactory: func(c cluster.FetcherConfig, latestConfigBlock *common.Block) cluster.BlockSource {
-			// TODO: update FetcherConfig from latestConfigBlock
+			// update FetcherConfig from latestConfigBlock
+			fc, err := cluster.UpdateFetcherConfigFromConfigBlock(c, latestConfigBlock)
+			bf_logger.Errorf("Could not update FetcherConfig fom Config Block: %v", err)
 			return &cluster.BlockPuller{
 				VerifyBlockSequence: verifyBlockSequence,
-				Logger:              flogging.MustGetLogger("orderer.common.cluster.puller").With("channel", support.ChannelID()),
+				Logger:              flogging.MustGetLogger("orderer.common.cluster.puller").With("channel", fc.Channel),
 				RetryTimeout:        clusterConfig.ReplicationRetryTimeout,
 				MaxTotalBufferBytes: clusterConfig.ReplicationBufferSize,
 				FetchTimeout:        clusterConfig.ReplicationPullTimeout,
```

Comment:
I think that the default `FetchTimeout` is 5 seconds. It means that every 5 seconds we will check for attestations by default... I'm not sure it's a good idea.

Reply:
We do not retry pulling attestations in case of timeout, but we retry pulling blocks where we may shuffle sources if needed.
Moreover I think we can have `FetchTimeOut` for __attestations__, some orders of magnitude smaller than `FetchTimeOut` for __blocks__. `FetchTimeOut` for _blocks_ in seconds and `FetchTimeout` for attestations in milliseconds.

Reply:
We should not have any timeout with a resolution of milliseconds. It's too small for internet